### ISSUE TITLE
404 not found when deploying

### ISSUE THREAD

**Issue by novakw267** [ga-wdi-boston/group-project#250]
After confirming our code was at a good and stable point on the local host we decided to deploy. Current with our deploy we're receiving a 404 not found after attempting to do anything on the site. 
front end config file
```
'use strict';

const config = {
  apiOrigins: {
    production: 'https://obscure-temple-20926.herokuapp.com/',
    development: 'http://localhost:4741'
  },
};

module.exports = config;
```

back end config database file
```
'use strict';

const mongooseBaseName = 'ScribbleBook';

const environment = process.env.NODE_ENV || 'development';
const database = {
  development: {
    mongoose: {
      name: `${mongooseBaseName}-development`,
      uri: `mongodb://localhost/${mongooseBaseName}-development`,
    },
  },
  test: {
    mongoose: {
      name: `${mongooseBaseName}-test`,
      uri: `mongodb://localhost/${mongooseBaseName}-test`,
    },
  },
};

module.exports = database[environment];
```
back end routes 
```
'use strict';

module.exports = require('lib/wiring/routes')

// create routes

// what to run for `GET /`
.root('root#root')

// standards RESTful routes
.resources('examples')

// users of the app have special requirements
.post('/sign-up', 'users#signup')
.post('/sign-in', 'users#signin')
.delete('/sign-out/:id', 'users#signout')
.patch('/change-password/:id', 'users#changepw')
.resources('users', { only: ['index', 'show'] })
.resources('blogs', { only: ['index','show', 'create','update', 'destroy'] })
.resources('pages', { only: ['create', 'index', 'show', 'update', 'destroy']});
// all routes created
```
heroku time line
```
![image](https://cloud.githubusercontent.com/assets/24447090/23775357/5413d494-04f6-11e7-94ea-2b5aab99b163.png)

```
heroku log 

![image](https://cloud.githubusercontent.com/assets/24447090/23775368/6012e730-04f6-11e7-8a1b-03f5ab27a49b.png)

 response from server
![image](https://cloud.githubusercontent.com/assets/24447090/23775387/71dcfb2c-04f6-11e7-9c82-4a3d416e0e9c.png)

Backend heroku server stuff
![image](https://cloud.githubusercontent.com/assets/24447090/23775409/890e5c46-04f6-11e7-8d48-b048a41da2fa.png)

*update*
we tried heroku log --tail and it didn't give any more of a detailed response than before, just so you know what we have tried. 
@ashlenprice  @wilsonkaya @squirtlesquad1988

**Comment by SquirtleSquad1988** [comment]
When attempting to log on to the mLab database form the backend terminal I got this error:

![image](https://cloud.githubusercontent.com/assets/24443003/23776561/080ef02c-04fd-11e7-86f8-702e49dd8cfc.png)

When I run:
```
mongo ds123050.mlab.com:23050/heroku_cj8j622q -u <dbuser> -p <dbpassword>
```

Obviously I replaced user and password fields with valid parameters.

**Comment by bernardlee** [comment]
Have you tried using cURL to sign up?  I just made a test account on your back end server with it.

**Comment by bernardlee** [comment]
The solution to this issue is in the original post.  Look at the images following "heroku log" and "response from server".

**Comment by AshlenPrice** [comment]
@bernardlee We looked at the images, the urls are the same not really sure what we're looking for, though. We're currently seeing if it's something to do with the //sign-up.

**Comment by jrhorn424** [comment]
Is `//sign-up` the same as `/sign-up`?

**Comment by SquirtleSquad1988** [comment]
We're gonna try that tomorrow but I've been comparing the console statements I get from our group-project app compared to my full-stack app and my full-stack app also has routes that end with //sign-up and //albums:

![image](https://cloud.githubusercontent.com/assets/24443003/23779158/b9d72a0e-050c-11e7-85ca-6b02f651bc62.png)

Would me not being able to access the mLab database be a symptom of this problem? Will try to remedy the issue how you suggested tomorrow and update.

**Comment by bernardlee** [comment]
> We're gonna try that tomorrow but I've been comparing the console statements I get from our group-project app compared to my full-stack app and my full-stack app also has routes that end with //sign-up and //albums:

One is Ruby on Rails and the other is Express.

> Would me not being able to access the mLab database be a symptom of this problem? Will try to remedy the issue how you suggested tomorrow and update.

It shouldn't be.  I was able to create an account on your deployed back end using cURL.

**Comment by raq929** [comment]
um... I see a route `//albums` on your get request. I suspect your actual route is `/albums`. This may or may not be causing the problem.

**Comment by raq929** [comment]
As a hint, I would try what we're suggesting first, before telling us that it won't work.

**Comment by novakw267** [comment]
That seemed to have worked as far as we can tell. 

It removed the / after the heroku url in the config.js file of our front end repo, so now it looks like this
```
'use strict';

const config = {
  apiOrigins: {
    production: 'https://obscure-temple-20926.herokuapp.com',
    development: 'http://localhost:4741'
  },
};

module.exports = config;
```
The double slash that was showing up in the URL path for the site, was interfering with the file path from the back end it seems. Removing it has fixed the issue as far as we can tell. 
 
With this now working as it is, it will create a clear deploy path later on when we update the site as we finalize more of the working features. 

I am going to close the issue for now but if there is more information I should include please let me know. 

TLDR: file paths are important, pay attention to those in your deploy error messages when something can't be found. 

Thank you for the help.
